### PR TITLE
Dfpl 872: Migrate Children for new 26 week extension fields

### PR DESCRIPTION
**Pull request by AlistairEdwinOsborne**
https://tools.hmcts.net/jira/browse/DFPL-872

Migration script to add new fields to ChildParty, including rollback + DFPL-89 Changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 1]
```diff
@@ -228,15 +235,60 @@ private void run826(CaseDetails caseDetails) {
...
+    private void run872Rollback(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
+        var caseId = caseData.getId();
+        List<Element<Child>> childrenInCase = caseData.getAllChildren();
+
+        Map<String, Object> caseDetailsData = caseDetails.getData();
+        if (isNotEmpty(childrenInCase)) {
+            log.info("Migration {id = DFPL-872-Rollback, case reference = {}} remove child extension fields", caseId);
+
+            List<Element<Child>> children = childrenInCase.stream()
```
Comment (jamesrferguson1) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:275]: wonder if we should maybe add a check that completionDate and extensionReason are non-null too?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 2]
```diff
@@ -228,15 +235,60 @@ private void run826(CaseDetails caseDetails) {
...
+        Map<String, Object> caseDetailsData = caseDetails.getData();
+        if (isNotEmpty(childrenInCase) && oldEightWeeksExtensionDate != null) {
+            log.info("Migration {id = DFPL-872, case reference = {}} extension date migration", caseId);
+
+            List<Element<Child>> children = childrenInCase.stream()
+                .map(element -> element(element.getId(),
+                     element.getValue().toBuilder()
+                        .party(element.getValue().getParty().toBuilder()
+                        .completionDate(oldEightWeeksExtensionDate)
+                        .extensionReason(CaseExtensionReasonList.TIMETABLE_FOR_PROCEEDINGS)
```
Comment (jamesrferguson1) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:253]: take it TIMETABLE_FOR_PROCEEDINGS was agreed as the default value?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java [hunk 3]
```diff
@@ -76,25 +80,57 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
...
+        List<Element<Child>> childrenInCase = caseData.getAllChildren();
+
+        Map<String, Object> caseDetailsData = caseDetails.getData();
+        if (isNotEmpty(childrenInCase)) {
+            log.info("Migration {id = DFPL-872-Rollback, case reference = {}} remove child extension fields", caseId);
+
+            List<Element<Child>> children = childrenInCase.stream()
+                .map(element -> element(element.getId(),
+                    element.getValue().toBuilder()
+                        .party(element.getValue().getParty().toBuilder()
```
Comment (DanCatchpole) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:124]: As far as I can tell this isn't clearing the extension fields on the children

Comment (DanCatchpole) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java:124]: May also be good to have a test or two for the rollback as well

**Review by DanCatchpole — APPROVED** [review]
lgtm! rollback changes should work

**Review by jamesrferguson1 — APPROVED** [review]

**Pull request merged**